orchestrator/src/fund.go

```diff
@@ -140,4 +142,31 @@ func (FundAction) RunMany(config Config, actionIOs []ActionIO) error {
 	return nil
 }
 
+func (FundAction) Validate(rawParams json.RawMessage) error {
+	var params FundParams
+	if err := json.Unmarshal(rawParams, &params); err != nil {
+		return fmt.Errorf("failed to unmarshal fund-keys params: %v", err)
+	}
+	fundKeysBaseDir := extractBaseDir(params.Prefix)
+	if pathExists(fundKeysBaseDir) {
+		return fmt.Errorf("path '%s' already exists. Please re-generate script using unique experiment name or different '-fund-keys-dir' CLI argument value", fundKeysBaseDir)
+	}
+	return nil
+}
+
+// Helper function to extract the first two levels of the path
+func extractBaseDir(prefix string) string {
+	parts := strings.Split(filepath.ToSlash(prefix), "/")
+	if len(parts) > 3 {
```

Comment:
Why not `> 1`? It'd be sufficient to check that the parent folder of prefix doesn't exist.

I believe with `>3` check `generated-keys/funding/itn08/key` won't be handled correctly (this is a pth you get from generator when using `-generate-privkeys 20` flag for generator

Reply:
I only considered the "fund-keys" action all this time...